pkg/gds/admin_test.go

```diff
@@ -694,6 +694,61 @@ func (s *gdsTestSuite) TestUpdateVASP() {
 	// TODO: Test common name and endpoint change is successful
 }
 
+func (s *gdsTestSuite) TestDeleteVASP() {
+	s.LoadFullFixtures()
+	defer s.ResetFullFixtures()
+
+	require := s.Require()
+	a := s.svc.GetAdmin()
+
+	deltaID := s.fixtures[vasps]["delta"].(*pb.VASP).Id
+	julietID := s.fixtures[vasps]["juliet"].(*pb.VASP).Id
+	xrayID := s.fixtures[certreqs]["xray"].(*models.CertificateRequest).Id
+	golfID := s.fixtures[vasps]["golfbucks"].(*pb.VASP).Id
+
+	// Attempt to delete a VASP that doesn't exist
+	request := &httpRequest{
+		method: http.MethodDelete,
+		path:   "/v2/vasps/invalid",
+		params: map[string]string{
+			"vaspID": "invalid",
+		},
+		claims: &tokens.Claims{
+			Email: "admin@example.com",
+		},
+	}
+	c, w := s.makeRequest(request)
+	rep := s.doRequest(a.DeleteVASP, c, w, nil)
+	s.APIError(http.StatusNotFound, "could not retrieve VASP record by ID", rep)
+
+	// VASP is in an invalid state for deletion
```

Comment:
Could we add a check to assert that delta is in the state we expect it to be? This is helpful to detect when fixtures change, but also tells us what state delta is in. 

Do you think we should test all states? We want to make really sure we do the right thing for deletes. I know that's excessive, but we are using `>` and `<` comparisons so maybe it's a good guard?

Reply:
Good point, I've pushed some changes to test all the states, mind taking another look?